lite/provider.go

```diff
@@ -30,3 +33,105 @@ type PersistentProvider interface {
 	// SaveFullCommit saves a FullCommit (without verification).
 	SaveFullCommit(fc FullCommit) error
 }
+
+// A provider that can update itself w/ more recent commit data.
+type UpdatingProvider interface {
+	Provider
+
+	// Update internal information by fetching information somehow.
+	// UpdateToHeight will block until the request is complete,
+	// or returns an error if the request cannot complete.
+	// Generally, one must call UpdateToHeight(h) before LatestFullCommit(_,h,h)
+	// will return this height.
+	//
+	// NOTE: Behavior with concurrent requests is undefined.  To make
+	// concurrent calls safe, look at the struct `ConcurrentUpdatingProvider`.
```

Comment:
There is no such struct `ConcurrentUpdatingProvider`. This probably refers to the (private) `concurrentProvider`.